### DISCUSSION TITLE
Anyone set up stig-manager with AWS RDS MySQL?

### DISCUSSION THREAD

**Discussion by petertirrell** [NUWCDIVNPT/stig-manager#1271] [Q&A]
We're testing out stig-manager and wanted to use Amazon's RDS MySQL flavor for the database backing.  The web app appears to be able to connect, but we're getting errors during what looks like the init setup of the database.

We are connecting using the default "admin" user that is created as part of the RDS setup. The error we are getting is along the lines of "failed to initialize database with file Access denied; you need (at least one of) the SUPER or SET_USER_ID privilege(s) for this operation".  Some more digging points to maybe RDS users not being able to even have "SUPER" permissions, and the default admin user isn't actually set as the DBA role.  And when we try and manage the user, to explicitly add the role, we get an error about not being able to GRANT options.  

Has anyone tried anything like this, or manually init the database with scripts for cases like this?  Thanks for any help!

**Comment by csmig** [comment]
Unfortunately, our [guidance in the documentation](https://stig-manager.readthedocs.io/en/latest/installation-and-setup/db.html#configure-mysql) was a bit confusing on this topic and has been updated. The MySQL account used by the API does NOT require the `SUPER` or `SET_USER_ID` privileges. And I have been told of at least one deployment of STIGMAN that uses the MySQL RDS service, but have no direct experience with it.

Please let us know if the following guidance is helpful:
To get MySQL ready for initial scaffolding of the STIG Manager schema, you need to prepare the MySQL instance by performing the following three tasks as a MySQL user with sufficient privileges to create schemas and users:

- create the STIGMAN database (schema) with suggested name 'stigman' 
  - Example: `CREATE DATABASE stigman`
- create a dedicated MySQL user to be used by the API (recommended) or skip if you want to use the privileged account 
  - Example: `CREATE USER 'stigman'@'%' IDENTIFIED BY 'new_password'`
- grant the dedicated MySQL user all privileges on the STIGMAN schema 
  - Example: `GRANT ALL ON stigman.* TO 'stigman'`

**Comment by tonyborelli-maker** [comment]
It's not too difficult to script these checks yourself, as I discuss in my blog, but I have tools available if you don't have the time that will work right out of the box without any tricky install/configuration steps. [How to STIG MySQL – Borelli Security Software Inc.](https://borellisecuritysoftware.com/blogs/mysql-8-0-stigs/how-to-stig-mysql)
